### PR TITLE
Fix column size estimation

### LINKED ISSUE #857: Column size estimation was not correct
Previously, column size is estimated by its `length`. However, length is not equivalent to size, e.g., `bigint(20)` has length 20 while its size is 8 bytes.

### PR DESCRIPTION
**Pull request by birdstorm**
Close #857 
Fix multiple incorrect assumptions on column size estimation.

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiIndexInfo.java [hunk 1]
```diff
@@ -40,9 +41,10 @@
   private final SchemaState schemaState;
   private final String comment;
   private final IndexType indexType;
-  private final long indexColumnLength;
   private final boolean isFakePrimaryKey;
 
+  private long indexColumnSize = 8;
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/meta/TiIndexInfo.java]: A comment is nice to have.

File: tikv-client/src/main/java/com/pingcap/tikv/types/BitType.java [hunk 2]
```diff
@@ -40,6 +40,15 @@ protected BitType(TiColumnInfo.InternalTypeHolder holder) {
     super(holder);
   }
 
+  @Override
+  public long getSize() {
+    if (isLengthUnSpecified()) {
+      return getDefaultDataSize();
+    } else {
+      return (getLength() + 7) / 8 + getPrefixSize();
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/types/BitType.java]: Does 7 and 9 have meaning?

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/types/BitType.java]: (x+7)/8 is equivalent to `ceil(x/8)`

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiIndexInfo.java (hunk 1 above)
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiIndexInfo.java]: perhaps i can delete this initialization....

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiIndexInfo.java]: done.

File: tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java [hunk 3]
```diff
@@ -46,7 +46,7 @@ private SetType(MySQLType tp) {
     for (int i = 0; i < 64; i++) {
       // In TiDB code base, it is taking complement of original value.
       // setIndexInvertValue[i] = ^setIndexValue[i] is invalid in Java.
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java]: comment also change?

File: tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java [hunk 4]
```diff
@@ -46,7 +46,7 @@ private SetType(MySQLType tp) {
...
-      tmpArr[i] = setIndexValue[i] ^ -1L;
+      tmpArr[i] = ~setIndexValue[i];
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java]: would u pls give a reason in comments?

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java [hunk 5]
```diff
@@ -44,7 +44,7 @@
   private final long maxColumnId;
   private final long maxIndexId;
   private final long oldSchemaId;
-  private final long columnLength;
+  private final long columnSize;
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java]: pls add comments, what `columnSize` means?

File: tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java [hunk 6]
```diff
@@ -119,39 +119,41 @@ protected DataType(
...
-      return MinInt16;
-    } else if (this.getType() == MySQLType.TypeInt24) {
-      return MinInt24;
-    } else if (this.getType() == MySQLType.TypeLong) {
-      return MinInt32;
-    } else if (this.getType() == MySQLType.TypeLonglong) {
-      return MinInt64;
-    } else {
-      throw new TypeException("Input Type is not a mysql SIGNED type");
+  Long signedLowerBound() throws TypeException {
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java]: `public` -> `protected`?
maybe better in a another PR (easy to cherry-pick)

File: tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java (hunk 4 above)
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java]: they are the same.

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java (hunk 5 above)
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java]: estimated column size.

File: tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java (hunk 3 above)
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java]: yes. i will delete these lines.

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java (hunk 5 above)
Comment (marsishandsome) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java]: pls add the comment in source code

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiTableInfo.java]: done.

File: tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java (hunk 3 above)
Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/types/SetType.java]: done.

**Pull request merged**